gpt4all-chat/tests/python/test_server_api.py

```diff
@@ -83,5 +116,132 @@ def chat_server(chat_server_config: dict[str, str]) -> Iterator[None]:
             raise CalledProcessError(retcode, process.args)
 
 
-def test_list_models_empty(chat_server: None) -> None:
-    assert request.get('models', wait=True) == {'object': 'list', 'data': []}
+@pytest.fixture
+def chat_server_with_model(chat_server_with_model_config: dict[str, str]) -> Iterator[None]:
+    chat_executable = Path(os.environ['CHAT_EXECUTABLE']).absolute()
+    with subprocess.Popen(chat_executable, env=chat_server_with_model_config) as process:
+        try:
+            yield
+        except:
+            process.kill()
+            raise
+        process.send_signal(signal.SIGINT)
+        if retcode := process.wait():
+            raise CalledProcessError(retcode, process.args)
+
+
+def test_with_models_empty(chat_server: None) -> None:
+    # non-sense endpoint
+    with pytest.raises(HTTPError) as excinfo:
+        request.get('foobarbaz', wait=True)
+    assert excinfo.value.response.status_code == 404
```

Comment:
In all these cases I think that the assertion belongs inside the context generated by `with` -- not sure how it will work otherwise.
```suggestion
        assert excinfo.value.response.status_code == 404
```

Reply:
No, this is a thing you can do - the result is not necessarily scoped within the context:
```
>>> from contextlib import nullcontext
>>> with nullcontext(1) as ctx:
...     pass
...
>>> ctx
1
```